### PR TITLE
Update workflows to leverage GitHub App

### PR DESCRIPTION
**Pull request by tuckerzp**
This adds the `triage-pr` as well as the `update-dependencies` workflows.

Part of closing https://github.com/EasyDynamics/easygrc/issues/44

### CODE REVIEW

File: .github/workflows/update-dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+
+      - name: Create update pull request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: Update NPM dependencies
+          title: Update NPM dependencies
+          body: >-
+            This was generated by the "${{ github.workflow }}" workflow, primarily
+            by running `node-check-updates` and `npm install`.
```
Comment (kylelaker) [inline .github/workflows/update-dependencies.yml:70]:
```suggestion
            by running `npm-check-updates` and `npm install`.
```

I understand this is probably incorrect in the original workflow too. Please open a PR to correct that as well.

File: .github/workflows/triage-pr.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            core.setOutput("review-decision", decision);
+            core.setOutput("is-draft", result.repository.pullRequest.isDraft);
+      - name: Generate token from App
+        # Only authenticate as the application if a review is still required. If the review
+        # criteria have been met then there's not much point in requesting another review and
+        # it will just generate unnecessary noise. Additionally, we really want to use the
+        # app's elevated permissions as infrequently as possible.
+        if: steps.pr-reviews.outputs.is-draft == 'false' && steps.pr-reviews.outputs.review-decision == 'REVIEW_REQUIRED'
+        id: generate_token
+        uses: tibdex/github-app-token@586e1a624db6a5a4ac2c53daeeded60c5e3d50fe
```
Comment (kylelaker) [inline .github/workflows/triage-pr.yml:72]: Can we look to see if there's a newer version of this for us to adopt?

**Review by laurelmay — COMMENTED** [review]
We probably should also change `.github/dependabot.yml` so that we don't get duplicate PRs.

File: .github/workflows/triage-pr.yml (hunk 2 above)
Comment (tuckerzp) [reply .github/workflows/triage-pr.yml]: [`f717b5ecd4534d3c4df4ce9b5c1c2214f0f7cd06`](https://github.com/tibdex/github-app-token/commit/f717b5ecd4534d3c4df4ce9b5c1c2214f0f7cd06) seems stable to me.

**Review by laurelmay — APPROVED** [review]

**Comment by mikeisen1** [comment]
Why is the `Build, Test, Deploy` check running again? Is there a way to rewrite the workflow such that the check only runs when the code itself changes and not when there is a review of the PR?

**Comment by kylelaker** [comment]
> Is there a way to rewrite the workflow such that the check only runs when the code itself changes and not when there is a review of the PR?

These are the triggers for that workflow. Note that PR reviews are not at all involved here. https://github.com/EasyDynamics/oscal-editor-deployment/blob/683f6ec50f045a8beb50d426c2eae16eb0894242/.github/workflows/container-test-deploy.yml#L2-L12

**Comment by tuckerzp** [comment]
>Why is the Build, Test, Deploy check running again?

@mikeisen1 it was running because of a new push to the branch, which should run the workflow.

**Review by mikeisen1 — APPROVED** [review]

**Review by laurelmay — APPROVED** [review]

**Comment by tuckerzp** [comment]
https://github.com/EasyDynamics/oscal-editor-deployment/runs/7618462116?check_suite_focus=true#step:8:194

Since this would be a somewhat appropriate place to do so, would we want to add a retry to avoid the chance of failure? I thought it was so unlikely we could just remove it but maybe I was wrong.

**Comment by kylelaker** [comment]
> Since this would be a somewhat appropriate place to do so, would we want to add a retry to avoid the chance of failure

Separate PR please

**Pull request merged**